bundles/core/src/main/java/com/adobe/cq/wcm/core/components/internal/models/v1/TableOfContentsImpl.java

```diff
@@ -0,0 +1,144 @@
+/*~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+ ~ Copyright 2022 Adobe
+ ~
+ ~ Licensed under the Apache License, Version 2.0 (the "License");
+ ~ you may not use this file except in compliance with the License.
+ ~ You may obtain a copy of the License at
+ ~
+ ~     http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~ Unless required by applicable law or agreed to in writing, software
+ ~ distributed under the License is distributed on an "AS IS" BASIS,
+ ~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~ See the License for the specific language governing permissions and
+ ~ limitations under the License.
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~*/
+package com.adobe.cq.wcm.core.components.internal.models.v1;
+
+import com.adobe.cq.export.json.ComponentExporter;
+import com.adobe.cq.export.json.ExporterConstants;
+import com.adobe.cq.wcm.core.components.models.TableOfContents;
+import com.adobe.cq.wcm.core.components.models.datalayer.ComponentData;
+import com.adobe.cq.wcm.core.components.util.AbstractComponentImpl;
+import com.day.cq.wcm.api.designer.Style;
+import org.apache.sling.api.SlingHttpServletRequest;
+import org.apache.sling.models.annotations.Exporter;
+import org.apache.sling.models.annotations.Model;
+import org.apache.sling.models.annotations.injectorspecific.InjectionStrategy;
+import org.apache.sling.models.annotations.injectorspecific.ScriptVariable;
+import org.apache.sling.models.annotations.injectorspecific.Self;
+import org.apache.sling.models.annotations.injectorspecific.ValueMapValue;
+import org.jetbrains.annotations.Nullable;
+
+import javax.annotation.PostConstruct;
+import java.util.Arrays;
+
+@Model(
+    adaptables = SlingHttpServletRequest.class,
+    adapters = {
+        TableOfContents.class,
+        ComponentExporter.class
+    },
+    resourceType = TableOfContentsImpl.RESOURCE_TYPE
+)
+@Exporter(name = ExporterConstants.SLING_MODEL_EXPORTER_NAME,
+    extensions = ExporterConstants.SLING_MODEL_EXTENSION)
+public class TableOfContentsImpl extends AbstractComponentImpl implements TableOfContents {
+
+    /**
+     * The resource type of the component associated with this Sling model.
+     */
+    public static final String RESOURCE_TYPE = "core/wcm/components/tableofcontents/v1/tableofcontents";
+
+    public static final String TOC_REQUEST_ATTR_FLAG = "cmp-toc__present";
```

Comment:
I wouldn't use the BEM notation for the request attribute, I'd go with `cmp-toc-present`

Reply:
Okay